src/gtp_context_reg.erl

```diff
@@ -215,14 +219,38 @@ maybe_init_rnd([]) ->
 maybe_init_rnd([{_, RndState}]) ->
     RndState.
 
-alloc_tei(_RndStateKey, _RndState, _KeyFun, 0) ->
+alloc_tei(_RndStateKey, _RndState, _KeyFun, 0, _) ->
     {error, no_tei};
-alloc_tei(RndStateKey, RndState0, KeyFun, Cnt) ->
-    {TEI, RndState} = rand:uniform_s(16#fffffffe, RndState0),
+alloc_tei(RndStateKey, RndState0, KeyFun, Cnt, #{prefix := Prefix0, length := PrefixLen} = Data) ->
```

Comment:
Data was called TeidCfg in the call. It is easier to understand if it keeps this name.